osu.Game/Overlays/News/Sidebar/MonthSection.cs

```diff
@@ -0,0 +1,180 @@
+                    {
+                        IsOpen = { BindTarget = IsOpen }
+                    },
+                    new PostsContainer
+                    {
+                        IsOpen = { BindTarget = IsOpen },
+                        Children = posts.Select(p => new PostButton(p)).ToArray()
+                    }
+                }
+            };
+        }
+
+        private class DropdownHeader : OsuClickableContainer
+        {
+            public readonly BindableBool IsOpen = new BindableBool();
+
+            private readonly SpriteIcon icon;
+
+            public DropdownHeader(int month, int year)
+            {
+                var date = new DateTime(year, month, 1);
+
+                RelativeSizeAxes = Axes.X;
+                Height = 15;
+                Action = IsOpen.Toggle;
+                Children = new Drawable[]
+                {
+                    new OsuSpriteText
+                    {
+                        Anchor = Anchor.CentreLeft,
+                        Origin = Anchor.CentreLeft,
+                        Font = OsuFont.GetFont(size: 12, weight: FontWeight.Bold),
+                        Text = date.ToString("MMM yyyy")
+                    },
+                    icon = new SpriteIcon
+                    {
+                        Anchor = Anchor.CentreRight,
+                        Origin = Anchor.CentreRight,
+                        Size = new Vector2(10),
+                        Icon = FontAwesome.Solid.ChevronDown
+                    }
+                };
+            }
+
+            protected override void LoadComplete()
+            {
+                base.LoadComplete();
+
+                IsOpen.BindValueChanged(open =>
+                {
+                    icon.Scale = new Vector2(1, open.NewValue ? -1 : 1);
+                }, true);
+            }
+        }
+
+        private class PostButton : OsuHoverContainer
+        {
+            protected override IEnumerable<Drawable> EffectTargets => new[] { text };
+
+            private readonly TextFlowContainer text;
+            private readonly APINewsPost post;
+
+            public PostButton(APINewsPost post)
+            {
+                this.post = post;
+
+                RelativeSizeAxes = Axes.X;
+                AutoSizeAxes = Axes.Y;
+                Child = text = new TextFlowContainer(t => t.Font = OsuFont.GetFont(size: 12))
+                {
+                    RelativeSizeAxes = Axes.X,
+                    AutoSizeAxes = Axes.Y,
+                    Text = post.Title
+                };
+            }
+
+            [BackgroundDependencyLoader]
+            private void load(OverlayColourProvider colourProvider, GameHost host)
+            {
+                IdleColour = colourProvider.Light2;
+                HoverColour = colourProvider.Light1;
+
+                TooltipText = "view in browser";
+                Action = () => host.OpenUrlExternally("https://osu.ppy.sh/home/news/" + post.Slug);
```

Comment:
Just a heads up that the news system is the only placed these domains are hard-coded. It's going to need to be fixed at some point to use `EndpointConfiguration`.

Reply:
Good point, although I expect that action will be going away at some point anyway, for I expect news to be ultimately shown in the client when the implementation is complete.

Reply:
Definitely, although I think we will still have "open on web" buttons like in user profile etc.

Anyways that's for another day, as these hard-coded urls are also in other classes this PR doesn't touch.